### PR TITLE
Update to new params cfeminter cfedistributor and cfevesting.

### PR DESCRIPTION
**Pull request by pablokulpa**

### CODE REVIEW

File: x/cfedistributor/types/message_update_params.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
+package types
+
+import (
+	"cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+)
+
```
Comment (pawel-borecki) [inline x/cfedistributor/types/message_update_params.go:9]: Unit tests for ValidateBasic?

File: x/cfeminter/keeper/migrations.go [hunk 2]
```diff
@@ -1,7 +1,7 @@
 package keeper
 
 import (
-	v110cfeminter "github.com/chain4energy/c4e-chain/x/cfeminter/migrations/v110"
+	v110cfeminter "github.com/chain4energy/c4e-chain/x/cfeminter/migrations/v2"
```
Comment (pawel-borecki) [inline x/cfeminter/keeper/migrations.go:4]: v110cfeminter to v2cfeminter

File: x/cfeminter/keeper/msg_server_update_params.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import (
+	"context"
+	"cosmossdk.io/errors"
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+
+	"github.com/chain4energy/c4e-chain/x/cfeminter/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
```
Comment (pawel-borecki) [inline x/cfeminter/keeper/msg_server_update_params.go:12]: Unit tests?

File: x/cfeminter/migrations/v2/params_test.go [hunk 4]
```diff
@@ -0,0 +1,262 @@
+package v2_test
+
```
Comment (pawel-borecki) [inline x/cfeminter/migrations/v2/params_test.go:2]: not git mv

File: x/cfeminter/types/message_update_params.go [hunk 5]
```diff
@@ -0,0 +1,74 @@
+package types
+
+import (
+	"cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
```
Comment (pawel-borecki) [inline x/cfeminter/types/message_update_params.go:8]: // ValidateBasic unit tests?

File: proto/c4echain/cfedistributor/tx.proto [hunk 6]
```diff
@@ -1,13 +1,63 @@
...
 
+import "cosmos/msg/v1/msg.proto";
+import "gogoproto/gogo.proto";
+import "cosmos_proto/cosmos.proto";
+import "c4echain/cfedistributor/sub_distributor.proto";
 // this line is used by starport scaffolding # proto/tx/import
 
 option go_package = "github.com/chain4energy/c4e-chain/x/cfedistributor/types";
 
 // Msg defines the Msg service.
```
Comment (pawel-borecki) [inline proto/c4echain/cfedistributor/tx.proto:12]: // Maybe we should add some rules about naming parameters related messages.
// If massage changes all params it should be named UpdateParams
// in other cases it should have suffix Params or Param
// It is to easily distinguish the the porpose if the message
// so here we would have
// UpdateParams
// UpdateSubDistributorParams
// UpdateSubDistributorDestinationShareParam
// UpdateSubDistributorBurnShareParam

File: proto/c4echain/cfedistributor/tx.proto [hunk 7]
```diff
@@ -1,13 +1,63 @@
...
 service Msg {
-    // this line is used by starport scaffolding # proto/tx/rpc
+  // this line is used by starport scaffolding # proto/tx/rpc
+  rpc UpdateAllSubDistributors(MsgUpdateAllSubDistributors) returns (MsgUpdateAllSubDistributorsResponse);
+  rpc UpdateSubDistributor(MsgUpdateSubDistributor) returns (MsgUpdateSubDistributorResponse);
+  rpc UpdateSubDistributorDestinationShare(MsgUpdateSubDistributorDestinationShare) returns (MsgUpdateSubDistributorDestinationShareResponse);
+  rpc UpdateMsgUpdateSubDistributorBurnShare(MsgUpdateSubDistributorBurnShare) returns (MsgUpdateSubDistributorBurnShareResponse);
```
Comment (pawel-borecki) [inline proto/c4echain/cfedistributor/tx.proto:18]: remove UpdateMsg from name

File: proto/c4echain/cfeminter/tx.proto [hunk 8]
```diff
@@ -1,13 +1,35 @@
...
+import "cosmos_proto/cosmos.proto";
+import "c4echain/cfeminter/params.proto";
+import "c4echain/cfeminter/minter.proto";
+
 // this line is used by starport scaffolding # proto/tx/import
 
 option go_package = "github.com/chain4energy/c4e-chain/x/cfeminter/types";
 
 // Msg defines the Msg service.
 service Msg {
```
Comment (pawel-borecki) [inline proto/c4echain/cfeminter/tx.proto:15]: // Don't see the point of separated request for changing denom without changing minter
// rather one message for changing all params
// and one message to change minters (with start time) as changing denom on working blockchain is not a common use case
// and also going with naming rules:
// MsgUpdateMinters should be changed to MsgUpdateMintersParams, MsgUpdateParams should be added, MsgUpdateMintDenom should be removed

File: proto/c4echain/cfevesting/tx.proto [hunk 9]
```diff
@@ -15,6 +18,7 @@ service Msg {
   rpc WithdrawAllAvailable(MsgWithdrawAllAvailable) returns (MsgWithdrawAllAvailableResponse);
   rpc CreateVestingAccount(MsgCreateVestingAccount) returns (MsgCreateVestingAccountResponse);
   rpc SendToVestingAccount(MsgSendToVestingAccount) returns (MsgSendToVestingAccountResponse);
+  rpc UpdateDenom(MsgUpdateDenom) returns (MsgUpdateDenomResponse);
```
Comment (pawel-borecki) [inline proto/c4echain/cfevesting/tx.proto:21]: if new rules will be applied the name should be UpdateParams or UpdateDenomParam

File: x/cfedistributor/keeper/msg_server_update_params.go [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+import (
+	"context"
+	"cosmossdk.io/errors"
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+
+	"github.com/chain4energy/c4e-chain/x/cfedistributor/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
```
Comment (pawel-borecki) [inline x/cfedistributor/keeper/msg_server_update_params.go:12]: unit tests?

File: x/cfevesting/keeper/migrations.go [hunk 11]
```diff
@@ -1,21 +1,35 @@
...
+func (m Migrator) Migrate1to2(ctx sdk.Context) error {
+	return v110cfevesting.MigrateStore(ctx, m.keeper.storeKey, m.keeper.cdc)
 }
 
 // Migrate2to3 migrates from version 2 to 3.
 func (m Migrator) Migrate2to3(ctx sdk.Context) error {
 	return v120cfevesting.MigrateStore(ctx, m.keeper.storeKey, m.keeper.cdc)
 }
+
+// Migrate2to3 migrates from version 3 to 4.
```
Comment (pawel-borecki) [inline x/cfevesting/keeper/migrations.go:32]: why 3 to 4 - 1.1.0 was 2, so 1.2.0 should be 3

File: x/cfevesting/types/message_update_denom.go [hunk 12]
```diff
@@ -0,0 +1,44 @@
+package types
+
+import (
+	"cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
```
Comment (pawel-borecki) [inline x/cfevesting/types/message_update_denom.go:8]: validate basic unit tests?

File: x/cfevesting/migrations/v4/keys.go [hunk 13]
```diff
@@ -0,0 +1,8 @@
+package v4
```
Comment (pawel-borecki) [inline x/cfevesting/migrations/v4/keys.go:1]: why v4 package?? current version is v3

File: x/cfevesting/module.go [hunk 14]
```diff
@@ -148,8 +153,10 @@ func (am AppModule) RegisterServices(cfg module.Configurator) {
 	types.RegisterQueryServer(cfg.QueryServer(), am.keeper)
 	types.RegisterMsgServer(cfg.MsgServer(), keeper.NewMsgServerImpl(am.keeper))
 
-	mig := keeper.NewMigrator(am.keeper)
-	err := cfg.RegisterMigration(types.ModuleName, 1, mig.Migrate2to3)
+	mig := keeper.NewMigrator(am.keeper, am.legacySubspace)
+	err := cfg.RegisterMigration(types.ModuleName, 1, mig.Migrate1to2)
+	err = cfg.RegisterMigration(types.ModuleName, 2, mig.Migrate2to3)
+	err = cfg.RegisterMigration(types.ModuleName, 3, mig.Migrate3to4)
```
Comment (pawel-borecki) [inline x/cfevesting/module.go:159]: current version should be 3

**Review by dawid-kruk — APPROVED** [review]

**Pull request merged**